From 2511925b9eb9b5cc2bb224128fb5f62af792bb37 Mon Sep 17 00:00:00 2001
From: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
Date: Fri, 4 Apr 2025 21:35:28 -0400
Subject: [PATCH] tests: use a temporary file:// as rendezvous

Before the change, the test case used a hardcoded port number to run the
tests. This made the test fail in some environments where the port
number was occupied (either by a different service or - more likely - by
a second run of the project test suite).

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
---
 tests/attr/test_data_parallel.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/attr/test_data_parallel.py b/tests/attr/test_data_parallel.py
index af904aba..a2564f45 100644
--- a/tests/attr/test_data_parallel.py
+++ b/tests/attr/test_data_parallel.py
@@ -4,6 +4,7 @@
 import copy
 import os
 from enum import Enum
+import tempfile
 from typing import Any, Callable, cast, Dict, List, Optional, Tuple, Type
 
 import torch
@@ -300,9 +301,8 @@ def _get_dp_attr_methods(
 if torch.cuda.is_available() and torch.cuda.device_count() != 0:
 
     # Distributed Data Parallel env setup
-    os.environ["MASTER_ADDR"] = "127.0.0.1"
-    os.environ["MASTER_PORT"] = "29500"
-    dist.init_process_group(backend="gloo", rank=0, world_size=1)
+    f = tempfile.NamedTemporaryFile(delete=False)
+    dist.init_process_group(backend="gloo", init_method=f"file://{f.name}", rank=0, world_size=1)
 
     class DataParallelTest(BaseTest, metaclass=DataParallelMeta):
         @classmethod
-- 
2.48.1

